mysql-test/suite/galera_3nodes/t/galera_ist_only.test

```diff
@@ -0,0 +1,164 @@
+--source include/galera_cluster_3nodes.inc
+
+SET @wsrep_sst_donor_global_saved = @@global.wsrep_sst_donor;
+--let $ofile= $MYSQLTEST_VARDIR/tmp/node.2.err
+
+--connection node_1
+CALL mtr.add_suppression(".*State transfer to .* failed: -125.*");
+CALL mtr.add_suppression(".*SST request is null, SST canceled.*");
+
+--connection node_3
+CALL mtr.add_suppression(".*State transfer to .* failed: -125.*");
+CALL mtr.add_suppression(".*SST request is null, SST canceled.*");
+
+--connection node_1
+CREATE TABLE t1 (a INT PRIMARY KEY);
+INSERT INTO t1 VALUES (1);
+
+--connection node_2
+--let $wait_condition = SELECT COUNT(*) = 1 FROM t1;
+--source include/wait_condition.inc
+
+# Case 1:
+# node_2 should catch up with the cluster after the restart
+# because both possible donors have all IST-needed information
+--source include/shutdown_mysqld.inc
+
+--connection node_1
+INSERT INTO t1 VALUES (2);
+INSERT INTO t1 VALUES (3);
+INSERT INTO t1 VALUES (4);
+
+--connection node_2
+--let $restart_parameters = "restart: --wsrep_provider_options='base_port=$NODE_GALERAPORT_2' --wsrep_sst_method=ist_only --log-error=$ofile"
+--let $do_not_echo_parameters = 1
+--source include/start_mysqld.inc
+--source include/galera_wait_ready.inc
+--let $assert_text = Table t1 should have 4 rows
+--let $assert_cond = COUNT(*) = 4 FROM t1;
+
+--let $assert_text = Assert that node_2 joined without SST
+--let $assert_file = $ofile
+--let $assert_select = The node will try to join the cluster using only IST.
+--let $assert_count = 1
+--source include/assert_grep.inc
+
+# Case 2:
+# node_2 should fail joining the cluster, because it doesn't have
+# grastate.dat file (cluster uuid is missing) 
+--source include/shutdown_mysqld.inc
+--remove_file $MYSQLTEST_VARDIR/mysqld.2/data/grastate.dat
+
+--connection node_1
+INSERT INTO t1 VALUES (5);
+INSERT INTO t1 VALUES (6);
+INSERT INTO t1 VALUES (7);
+
+--connection node_2
+--let $restart_parameters = "restart: --wsrep_provider_options='base_port=$NODE_GALERAPORT_2' --wsrep_sst_method=ist_only --log-error=$ofile"
+--let $expect_crash_on_start = 1
+--let $do_not_echo_parameters = 1
+--source include/start_mysqld.inc
+
+# node_2 can recover only via full SST
+--let $restart_parameters = "restart: --wsrep_provider_options='base_port=$NODE_GALERAPORT_2'"
+--let $expect_crash_on_start = 0 
+--let $do_not_echo_parameters = 1
+--source include/start_mysqld.inc
+--source include/galera_wait_ready.inc
+--let $assert_text = Table t1 should have 7 rows
+--let $assert_cond = COUNT(*) = 7 FROM t1;
+
+--let $assert_text = Assert that node_2 failed to join without SST
+--let $assert_file = $ofile
+--let $assert_select = Local state UUID 00000000-0000-0000-0000-000000000000 is different from group state
+--let $assert_count = 1
+--source include/assert_grep.inc
+
+# Case 3:
+# node_2 should fail joining the cluster, because the requested donor
+# cannot serve IST
+--connection node_1
+SET @@global.wsrep_provider_options = "pc.weight=3";
```

Comment:
Why is this required when we are setting `wsrep_desync=1` to make node_3 serve as donor?